lib/components/molecules/Table/Table.js

```diff
@@ -0,0 +1,59 @@
+import React, { cloneElement, Children } from 'react';
+import styled from 'styled-components';
+
+import styles from './Table.style';
+// eslint-disable-next-line flowtype/no-types-missing-file-annotation
+import type { Props, ColumnProps } from './types';
+
+// eslint-disable-next-line flowtype/no-types-missing-file-annotation
+const Table = ({ children, data, className, ...props }: Props) => {
+  return (
+    <table className={className} {...props} role="table">
+      <thead>
+        <tr role="row">
+          {Children.map(children, column => (
+            <th key={column.props.name} role="columnheader">
+              {cloneElement(column, { intent: 'header' })}
+            </th>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {data.map(item => (
+          <tr key={item.id} role="row">
+            {Children.map(children, column => (
```

Comment:
Don't like this nested loop, we should have broken this complicity by creating a smaller component.